packages/vscode-extension/src/network/App.tsx

```diff
@@ -10,15 +10,31 @@ import { useNetwork } from "./providers/NetworkProvider";
 
 function App() {
   const networkLogContainerRef = useRef<HTMLDivElement | null>(null);
+  const networkDetailsContainerRef = useRef<HTMLDivElement>(null);
+  const networkLogDetailsSize = useRef<string>("50%");
+
   const [networkLogContainerHeight, setNetworkLogContainerHeight] = useState<number | undefined>(
     networkLogContainerRef?.current?.clientHeight
   );
 
+  const { networkLogs } = useNetwork();
+
+  const [selectedNetworkLogId, setSelectedNetworkLogId] = useState<string | null>(null);
+
+  const selectedNetworkLog = useMemo(() => {
+    const fullLog = networkLogs.find((log) => log.requestId === selectedNetworkLogId);
+    if (!fullLog) {
+      setSelectedNetworkLogId(null);
+    }
+    return fullLog || null;
```

Comment:
kinda weird to have a type guard a line before and then do `|| null`. this code doesn't feel right